src/agent/rustjail/src/cgroups/systemd/cgroups_path.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+
+use super::common::{DEFAULT_SLICE, SCOPE_SUFFIX, SLICE_SUFFIX};
+use std::string::String;
+
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct CgroupsPath {
+    pub slice: String,
+    pub prefix: String,
+    pub name: String,
+}
+
+impl CgroupsPath {
+    pub fn new(cgroups_path_str: &str) -> Result<Self> {
+        let path_vec: Vec<&str> = cgroups_path_str.split(':').collect();
```

Comment:
It's better to check the path format, at least it must have 3 parts, to avoid panic of out of index error.